code/modules/mining/ores_coins.dm

```diff
@@ -6,6 +6,11 @@
 	icon_state = "ore"
 	var/points = 0 //How many points this ore gets you from the ore redemption machine
 	var/refined_type = null //What this ore defaults to being refined into
+	var/reagent_id = null
+
+/obj/item/weapon/ore/on_smelt()
+	if(reagent_id)

```

Comment:
This can be just `return reagent_id`, since this would return null regardless of id availability.